src/frontend/src/utils/requestUtil.ts

```diff
@@ -0,0 +1,23 @@
+import { BASE_URL } from '../constant/api';
+
+export const client = async (
+  endpoint: string,
+  method = 'POST',
+  body = {},
+  headers = {},
+  customConfig = {},
+) => {
+  const options = {
+    method: method,
+    headers: { 'Content-Type': 'application/json', ...headers }, // by default setting the content-type to be json type
+    body: body ? JSON.stringify(body) : null,
```

Comment:
since you have defined `body` with default = `{}`, won't `body ? ` always be true? Did you give it a try with GET requests?

Reply:
good catch, i will remove the default value for the body